alphadia/workflow/managers/raw_file_manager.py

```diff
@@ -0,0 +1,151 @@
+"""Manager handling the raw data file and its statistics."""
+
+import logging
+import os
+
+import numpy as np
+
+from alphadia.data import alpharaw_wrapper, bruker
+from alphadia.workflow.config import Config
+from alphadia.workflow.manager import BaseManager
+
+logger = logging.getLogger()
+
+
+class RawFileManager(BaseManager):
+    def __init__(
+        self,
+        config: None | Config = None,
+        path: None | str = None,
+        load_from_file: bool = False,
+        **kwargs,
+    ):
+        """Contains and updates timing information for the portions of the workflow."""
+        self.stats = {}  # needs to be before super().__init__ to avoid overwriting when loading
+
+        super().__init__(path=path, load_from_file=load_from_file, **kwargs)
+        self.reporter.log_string(f"Initializing {self.__class__.__name__}")
+        self.reporter.log_event("initializing", {"name": f"{self.__class__.__name__}"})
+
+        self._config: Config = config
+
+    def get_dia_data_object(
+        self, dia_data_path: str
+    ) -> bruker.TimsTOFTranspose | alpharaw_wrapper.AlphaRaw:
+        """Get the correct data class depending on the file extension of the DIA data file.
+
+        Parameters
+        ----------
+
+        dia_data_path: str
+            Path to the DIA data file
+
+        Returns
+        -------
+        typing.Union[bruker.TimsTOFTranspose, thermo.Thermo],
+            TimsTOFTranspose object containing the DIA data
+
+        """
+        file_extension = os.path.splitext(dia_data_path)[1]
+
+        is_wsl = self._config["general"]["wsl"]
+        if is_wsl:
+            # copy file to /tmp # TODO why is that?
```

Comment:
There was some weird behjaviour in WSL.
I think I will actually drop WSL support soon, as it's not needed anymore.

Reply:
I am happy to drop it right away?

Reply:
Please do! But it’s across the GUI, CLI etc. so I would do it in a separate PR